### PR TITLE
Partial centroid size adjustment for GPU K-Means

### PR DESCRIPTION
**Pull request by amgrigoriev**

### CODE REVIEW

**Review by michael-smirnov — COMMENTED** [review]
Is it a performance optimization or a bugfix?

**Comment by amgrigoriev** [comment]
The goal is to fit in memory for huge workloads (bug fix)

File: cpp/daal/src/algorithms/kmeans/oneapi/kmeans_dense_lloyd_kernel_base_ucapi_impl.i [hunk 1]
```diff
@@ -563,6 +560,20 @@ Status KMeansDenseLloydKernelBaseUCAPI<algorithmFPType>::getBlockSize(uint32_t n
     return Status();
 }
 
+template <typename algorithmFPType>
+Status KMeansDenseLloydKernelBaseUCAPI<algorithmFPType>::fitPartialCentroidSize(uint32_t nClusters, uint32_t nFeatures)
+{
+    while (_nPartialCentroids > _nValuesInBlock / nClusters / nFeatures)
```
Comment (KulikovNikita) [inline cpp/daal/src/algorithms/kmeans/oneapi/kmeans_dense_lloyd_kernel_base_ucapi_impl.i:566]:
```suggestion
    while (_nPartialCentroids * nClusters * nFeatures > _nValuesInBlock)
```

**Review by KulikovNikita — APPROVED** [review]

**Review by mchernov-intel — APPROVED** [review]

**Pull request merged**